wasm-schema-sandbox/src/lib.rs

```diff
@@ -0,0 +1,213 @@
+use ion_schema::authority::{DocumentAuthority, MapDocumentAuthority};
+use ion_schema::external::ion_rs::value::owned::OwnedElement;
+use ion_schema::external::ion_rs::value::reader::{element_reader, ElementReader};
+use ion_schema::external::ion_rs::IonResult;
+use ion_schema::result::{IonSchemaResult, ValidationResult};
+use ion_schema::schema::Schema;
+use ion_schema::system::SchemaSystem;
+use ion_schema::types::TypeRef;
+use std::rc::Rc;
+use std::str;
+use wasm_bindgen::prelude::*;
+
+extern crate web_sys;
+
+// A macro to provide `println!(..)`-style syntax for `console.log` logging.
+macro_rules! log {
+    ( $( $t:tt )* ) => {
+        web_sys::console::log_1(&format!( $( $t )* ).into());
+    }
+}
+
+// Verify if the given value is valid and print violation for invalid value
+fn check_value(value: OwnedElement, type_ref: &TypeRef) -> ValidationResult {
+    type_ref.validate(&value)
+}
```

Comment:
I think `type_ref.validate(&value)` is pretty clear, what's the motivation for the wrapper function?

Reply:
I initially created that to do print validation result but since now I am just saving it in the `SchemaValidationResult` this can be eliminated.